tasks/autorestic.yml

```diff
@@ -1,34 +1,61 @@
 ---
 # autorestic tasks file for autorestic
+- name: Ensure autorestic directory exist
+  ansible.builtin.file:
+    state: directory
+    path: /opt/autorestic/bin
+    mode: 0755
+    owner: root
+
 - name: "Download autorestic binary v{{ autorestic_version }} for aarch64"
   # E.g., Raspberry Pi 4
   ansible.builtin.get_url:
     url: "https://github.com/cupcakearmy/autorestic/releases/download/v{{ autorestic_version }}/autorestic_{{ autorestic_version }}_linux_arm64.bz2" # noqa yaml[line-length]
     dest: /tmp/autorestic.bz2
-    mode: 0755
   when: ansible_architecture == "aarch64"
 
 - name: "Download autorestic binary v{{ autorestic_version }} for armv7l"
   # E.g., Raspberry Pi 3 (32bit)
   ansible.builtin.get_url:
     url: "https://github.com/cupcakearmy/autorestic/releases/download/v{{ autorestic_version }}/autorestic_{{ autorestic_version }}_linux_arm.bz2"
     dest: /tmp/autorestic.bz2
-    mode: 0755
   when: ansible_architecture == "armv7l"
 
 - name: "Download autorestic binary v{{ autorestic_version }} for x86_64"
   ansible.builtin.get_url:
     url: "https://github.com/cupcakearmy/autorestic/releases/download/v{{ autorestic_version }}/autorestic_{{ autorestic_version }}_linux_amd64.bz2" # noqa yaml[line-length]
     dest: /tmp/autorestic.bz2
-    mode: 0755
   when: ansible_architecture == "x86_64"
 
-- name: Extract and install autorestic # noqa no-changed-when
+- name: "Download autorestic binary v{{ autorestic_version }} for other architectures"
+  # Generic list containing available linux builds
+  ansible.builtin.get_url:
+    url: "https://github.com/cupcakearmy/autorestic/releases/download/v{{ autorestic_version }}/autorestic_{{ autorestic_version }}_linux_{{ item }}.bz2"
+    dest: /tmp/autorestic.bz2
+  when: ansible_architecture == item
+  with_items:
+    - mips
+    - mipsle
+    - mips64
+    - mips64le
+    - ppc64le
+    - s390x
+    - 386
```

Comment:
This is pretty cool :) - So it will run the action only for the matching item in `ansible_architecture`?

I usually see `with_items` for looping over each, but this is slightly different? 🙂

Reply:
the difference really is the `when` condition. as i understand it, `with_items` loops the whole task, so the condition is tested on every single loop run, hence the possibility to use it for our case.